controllers/v2/volunteer/api.py

```diff
@@ -0,0 +1,29 @@
+from flask_restful import Resource, marshal_with, reqparse
+
+from controllers.v2.v2_blueprint import v2_bp, v2_api
+from controllers.v2.volunteer.response_models import volunteer_personal_info
+from domain import session_scope, UserType
+from repository.volunteer_repository import get_volunteer_info
+from services.jwk import has_role
+
+parser = reqparse.RequestParser()
+parser.add_argument('user_id', action='store', type=str)
+
+
+class VolunteerInfo(Resource):
+    """
+    Get the personal detail for a specific volunteer
+    This information should only be visible for the supervisor and volunteer themselves
+    """
+
+    @marshal_with(volunteer_personal_info)
+    @has_role(UserType.ROOT_ADMIN)
```

Comment:
Should users be able to access information about themselves? Maybe this could be changed to allow any authenticated user to query, but add a check that prevents non-supervisors from viewing information about anyone but themselves. If you make the change I suggest below, this should be pretty easy.

Reply:
This can be a stretch goal for a different sprint